### PR TITLE
Support https imports.

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by NewLunarFire** [comment]
What's left should simply be to add the https connector when instantiating a client for fecthing. i'd would like to rewrite parts of deno_dir.rs (code_fetch, resolve_module, is_remote) using PathBufs and Urls so we can match on the scheme part of the path.

That, or we keep using strings comparaisons (starts_with("http://") or starts_with("https://"))

**Comment by ry** [comment]
@NewLunarFire Refactoring deno_dir.rs to pass around URLs instead of strings sounds good. It should be done independently of adding https support tho.

File: build_extra/rust/BUILD.gn [hunk 1]
```diff
@@ -580,3 +581,231 @@ rust_crate("tokio_threadpool") {
...
+    "$ring_root/include/GFp/type_check.h",
+    "$ring_root/third_party/fiat/curve25519.c",
+    "$ring_root/third_party/fiat/curve25519_tables.h",
+    "$ring_root/third_party/fiat/internal.h",
+
+    #"$ring_root/crypto/fipsmodule/modes/polyval.c",
+  ]
+  if (is_mac) {
+    sources += [
+      "$ring_root/pregenerated/aes-586-macosx.S",
```
Comment (piscisaureus) [inline build_extra/rust/BUILD.gn]: Did you compile these or were they in the crate?

File: libdeno/file_util_test.cc [hunk 2]
```diff
@@ -38,7 +38,7 @@ TEST(FileUtilTest, ExePath) {
   EXPECT_TRUE(exe_path.find("/") == 0 || exe_path.find(":\\") == 1);
   // FIlename is the name of the test binary.
   std::string exe_filename = deno::Basename(exe_path);
-  EXPECT_EQ(exe_filename.find("test_cc"), 0);
```
Comment (piscisaureus) [inline libdeno/file_util_test.cc]: Sorry. I don't get this warning locally.
I wish the CI would highlight new warnings and post them as github comments.

File: build_extra/rust/BUILD.gn [hunk 3]
```diff
@@ -7,6 +7,7 @@
 # Use //tools/sync_third_party.py instead of running "cargo install".
 
 import("rust.gni")
+import("//build/compiled_action.gni")
```
[the remark this answers is not in the document]
Comment (ry) [reply build_extra/rust/BUILD.gn]: Yes - removed.

File: libdeno/file_util_test.cc (hunk 2 above)
Comment (ry) [reply libdeno/file_util_test.cc]: np

File: build_extra/rust/BUILD.gn (hunk 1 above)
Comment (ry) [reply build_extra/rust/BUILD.gn]: In the crate.

File: src/fs.rs [hunk 4]
```diff
@@ -23,8 +23,9 @@ pub fn make_temp_dir(
     Some(ref p) => p.to_path_buf(),
     None => std::env::temp_dir(),
   }.join("_");
+  let mut rng = rand::thread_rng();
```
Comment (piscisaureus) [inline src/fs.rs]: 🤷🏻‍♂️but out of curiosity, this change? From the docs:

```
Retrieve the lazily-initialized thread-local random number generator, seeded by the system. Intended to be used in method chaining style, e.g. thread_rng().gen::<i32>().
```

Comment (ry) [reply src/fs.rs]: I had to downgrade rand to be compatible with other deps and the API is different.

Comment (piscisaureus) [reply src/fs.rs]: Ah ok got it.

Comment (ry) [reply src/fs.rs]: Just realized I didn't change the version in Cargo.toml... Fixed.

**Pull request merged**